packages/terra-list/src/ListSectionHeader.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import Arrange from 'terra-arrange';
+import styles from './ListSectionHeader.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+};
+
+const propTypes = {
+  /**
+   * Text to be displayed on the ListSectionHeader.
+   */
+  title: PropTypes.string,
+  /**
+   * Callback function triggered when the accordion icon is clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Whether the accordion icon should be displayed in its 'open' or 'closed' position.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=2`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+};
+
+const defaultProps = {
+  title: '',
+  onClick: undefined,
+  isOpen: false,
+  level: 2,
+};
+
+const isRecognizedKeyPress = event => ((event.nativeEvent.keyCode === KEYCODES.ENTER) || (event.nativeEvent.keyCode === KEYCODES.SPACE));
+
+class ListSectionHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { isActive: false };
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.wrapOnKeyUp = this.wrapOnKeyUp.bind(this);
+  }
+
+  wrapOnKeyDown(onKeyDown) {
+    return ((event) => {
+      if (isRecognizedKeyPress(event) && !this.state.isActive) {
+        this.setState({ isActive: true });
+
+        // Call the same function the user provides for a click event
+        if (this.props.onClick) {
+          this.props.onClick(event);
+        }
+      }
+
+      if (onKeyDown) {
+        onKeyDown(event);
+      }
+    });
+  }
+
+  wrapOnKeyUp(onKeyUp) {
+    return ((event) => {
+      if (isRecognizedKeyPress(event) && this.state.isActive) {
+        this.setState({ isActive: false });
+      }
+
+      if (onKeyUp) {
+        onKeyUp(event);
+      }
+    });
+  }
+
+  render() {
+    const {
+      title,
+      onClick,
+      isOpen,
```

Comment:
Do we want this to be a controlled component? `isOpen` is currently just a prop.

Reply:
I have created this since I would be using this prop in [Accordion Section Header](https://github.com/cerner/terra-core/blob/Accordion-Section-Header/packages/terra-list/src/Section.jsx#L35). We have just divided the entire workflow into 2 parts since it would be a massive PR if we combine both of them.